pkg/flowkit/services/accounts.go

```diff
@@ -276,19 +278,70 @@ func (a *Accounts) Create(
 	return a.gateway.GetAccount(*newAccountAddress[0]) // we know it's the only and first event
 }
 
+// Contract defines properties of a contract like name of the contract,
+// source code, possible init arguments, the filename and network are only
+// required if a contract has imports that need resolving.
+type Contract struct {
+	Name     string
+	Source   []byte
+	Args     []cadence.Value
+	Filename string
+	Network  string
+}
+
+func (c *Contract) validate(hasImports bool) error {
+	if c.Source == nil {
+		return fmt.Errorf("must provide contract source code")
+	}
+	if c.Name == "" {
+		return fmt.Errorf("must provide contract name")
+	}
+	if hasImports && c.Network == "" {
+		return fmt.Errorf("missing network, specify which network to use to resolve imports in transaction code")
+	}
+	if hasImports && c.Filename == "" {
+		return fmt.Errorf("cannot resolve imports without specifying a contract filename")
+	}
+	return nil
+}
+
 // AddContract deploys a contract code to the account provided with possible update flag.
 func (a *Accounts) AddContract(
 	account *flowkit.Account,
-	contractName string,
-	contractSource []byte,
+	contract *Contract,
 	updateExisting bool,
-	contractArgs []cadence.Value,
 ) (*flow.Account, error) {
+	resolver, err := contracts.NewResolver(contract.Source)
+	if err != nil {
+		return nil, err
+	}
+
+	err = contract.validate(resolver.HasFileImports())
```

Comment:
```suggestion
        hasFileImports = resolver.HasFileImports()
        
	err = contract.validate(hasFileImports)
```

I can't remember what the cost is of calling `HasFileImports()` , but might make sense to save to a variable here.